### DISCUSSION TITLE
Close modal when clicking outside

### DISCUSSION THREAD

**Discussion by Gw3i** [neuefische/questions#135] [Web] [answered]
Hey everybody,
I would like to close my modal when clicking outside of it.
There is some information on the web, but I couldn't find a good solution for it.

I've tried this one but, it's not working because of this error: **Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?**
<img width="1427" alt="Screenshot 2022-09-04 at 13 05 19" src="https://user-images.githubusercontent.com/101993818/188310154-780e14e4-e894-4e53-996d-f00e1f7cce0a.png">
```javascript

function useOnClickOutside(ref, handler) {
		useEffect(() => {
			const listener = event => {
				if (!ref.current || ref.current.contains(event.target)) {
					return;
				}
				handler(event);
			};
			document.addEventListener('mousedown', listener);
			document.addEventListener('touchstart', listener);
			return () => {
				document.removeEventListener('mousedown', listener);
				document.removeEventListener('touchstart', listener);
			};
		}, [ref, handler]);
	}

useOnClickOutside(ref, () => setIsShown(false));
```

Here is my code: https://github.com/Gw3i/capstone-project/pull/39/
Here is my app: https://capstone-project-a9t9w8jtt-gw3i.vercel.app/create-account (you need to create an account and login. The search for a YouTube channel, choose it, choose a playlist and try to add a video. The modal pops up)

Does someone has a good solution?

Thanks!

**Comment by JessicaLoers** [comment, marked as answer]
Hello Vlady 
here is an alternative more simple solution without using `useRef()` 

👉 [Codesanbox Modal-Example](https://codesandbox.io/s/modal-example-github-discussion-135-kqtezx?file=/src/App.js)

**Reply by Gw3i** [reply]
So easy 😅 Thanks!
